src/matlab.ts

```diff
@@ -1,40 +1,112 @@
-// Copyright 2022-2023 The MathWorks, Inc.
+// Copyright 2022-2024 The MathWorks, Inc.
 
 import * as core from "@actions/core";
+import * as exec from "@actions/exec";
 import * as http from "@actions/http-client";
 import * as io from "@actions/io";
 import * as tc from "@actions/tool-cache";
 import * as fs from "fs";
+import { homedir } from "os";
+import * as path from "path";
 import properties from "./properties.json";
-import * as script from "./script";
 
 export interface Release {
     name: string;
     version: string;
     update: string;
 }
 
-export async function makeToolcacheDir(release: Release): Promise<[string, boolean]> {
+export async function makeToolcacheDir(release: Release, platform: string): Promise<[string, boolean]> {
     let toolpath: string = tc.find("MATLAB", release.version);
     let alreadyExists = false;
     if (toolpath) {
         core.info(`Found MATLAB ${release.name} in cache at ${toolpath}.`);
         alreadyExists = true;
     } else {
-        fs.writeFileSync(".keep", "");
-        toolpath = await tc.cacheFile(".keep", ".keep", "MATLAB", release.version);
-        io.rmRF(".keep");
+        toolpath = await windowsToolpath(release, platform) || await defaultToolpath(release, platform);
     }
     return [toolpath, alreadyExists]
 }
 
-export async function setupBatch(platform: string) {
-    const batchInstallDir = script.defaultInstallRoot(platform, "matlab-batch")
-    await script
-        .downloadAndRunScript(platform, properties.matlabBatchUrl, [batchInstallDir])
-        .then(() => {
-            core.addPath(batchInstallDir);
-        });
+async function windowsToolpath(release: Release, platform: string): Promise<string | false> {
+    if (platform !== "win32" ) {
+        return false
+    }
+
+    // only apply optimization for github hosted runners
```

Comment:
Comment seems to be slightly different from logic

Reply:
The logic is aligned with the comment because it's a bail out condition if you're not on a GitHub hosted runner. I updated and flipped the polarity to make it clearer.